src/grid/grid.directive.ts

```diff
@@ -0,0 +1,86 @@
+import { Directive, HostBinding, Input, OnInit } from '@angular/core';
```

Comment:
Hm, if we're doing everything in one file I'd be tempted to call it `grid.directives.ts` ... unless that would add confusion?

Generally things have been one component/directive per file, but I'm not opposed to a multi-directive file if it makes sense